### PR TITLE
RavenDB-14851 remove time-series replication optimizations

### PR DESCRIPTION
**Pull request by karmeli87**
The work on the rollup/retention revealed some replication issues.
This PR hopefully will bring the replication to a correct and stable state for all cases.

### CODE REVIEW

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs [hunk 1]
```diff
@@ -176,6 +208,23 @@ internal async Task HandleChanges()
 
                         await AddNewPolicy(collectionName, prev, policy.Policy);
                     }
+
+                    //TODO: clean up the removed policies from the rollups table
```
Comment (ayende) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs]: Aren't they going to be cleaned up automatically when we read them, see they are not relevant and removed it?

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 2]
```diff
@@ -374,96 +375,131 @@ bool TryRemoveRange(ref TableValueReader reader, out Table.TableValueHolder next
...
 
-                void RemoveTimeSeriesNameFromMetadata(DocumentsOperationContext ctx, string docId, string tsName)
+                void Notify()
                 {
-                    var doc = _documentDatabase.DocumentsStorage.Get(ctx, docId);
-                    if (doc == null)
-                        return;
+                    context.Transaction.AddAfterCommitNotification(new TimeSeriesChange
+                    {
+                        ChangeVector = changeVector, // change-vector can be null, if we delete an entire segments
```
Comment (ayende) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Why? Shouldn't we still generate a change vector for the deletion?

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs]: right, done.

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Since the optimization was removed, so it that comment.

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
@karmeli87 can you check failing test here: https://jenkins.hibernatingrhinos.com/job/PR_Tests_v5.0/730/

**Pull request merged**